pkg/autorelay/api/blacklist.go

```diff
@@ -33,6 +34,19 @@ func (h *RelayServerHandler) AddBlacklist(c echo.Context) (err error) {
 		return rumerrors.NewInternalServerError(err)
 	}
 
+	relay := h.node.GetRelay()
+	if relay != nil {
+		from, err := peer.Decode(param.FromPeer)
+		if err != nil {
+			return rumerrors.NewBadRequestError(err.Error())
+		}
+		to, err := peer.Decode(param.ToPeer)
+		if err != nil {
+			return rumerrors.NewBadRequestError(err.Error())
+		}
+		relay.DisconnectByPeerID(from, to)
```

Comment:
The libp2p connection It's a bidirectional connection, and is it need call an extra relay.DisconnectByPeerID(to, from)?

Reply:
yeah true

Reply:
nah, either side is closed, the other side will be closed too, no need to call it twice. that means, `DisconnectByPeerID(from, to)` is the same as `DisconnectByPeerID(to, from)` , just affects from which node to disconnect

but things are different during the connection establishment phase, a connection can be requested by both two peers, so if we want to blacklist a connection, we have to check both direction, so we call `CheckBlacklist` twice [here](https://github.com/rumsystem/quorum/pull/193/files#diff-7f67a0b91ba6efff251c250d37614b593f2e461ef140eee7001707d3d8bec914R47)